lib/index.js

```diff
@@ -0,0 +1,287 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+// const env = require('./env');
+const forge = require('node-forge');
+const {util: {binary: {base58}}} = forge;
+const ec = new require('elliptic').ec('secp256k1');
+const util = require('./util');
+
+class Secp256KeyPair {
+  /**
+   * @param {KeyPairOptions} options - The options to use.
+   * @param {string} options.id - The key ID.
+   * @param {string} options.controller - The key controller.
+   * @param {string} options.publicKeyBase58 - Base58 encoded Public Key
+   * unencoded is 32-bytes.
+   * @param {string} options.privateKeyBase58 - Base58 Private Key
+   * unencoded is 64-bytes.
+   */
+  constructor(options = {}) {
+    this.id = options.id;
+    this.controller = options.controller;
+    this.type = 'EcdsaSecp256k1VerificationKey2019';
+    this.privateKeyBase58 = options.privateKeyBase58;
+    this.publicKeyBase58 = options.publicKeyBase58;
+  }
+
+  /**
+   * Returns the Base58 encoded public key.
+   *
+   * @returns {string} The Base58 encoded public key.
+   */
+  get publicKey() {
+    return this.publicKeyBase58;
+  }
+
+  /**
+   * Returns the Base58 encoded private key.
+   *
+   * @returns {string} The Base58 encoded private key.
+   */
+  get privateKey() {
+    return this.privateKeyBase58;
+  }
+
+  /**
+   * Generates a KeyPair with an optional deterministic seed.
+   * @param {KeyPairOptions} [options={}] - The options to use.
+   *
+   * @returns {Promise<Secp256KeyPair>} Generates a key pair.
+   */
+  static async generate(options = {}) {
+
+    const key = ec.genKeyPair();
+    const pubPoint = key.getPublic();
+    // pubPoint.encode returns byte array '04' + x +  y;
+    const publicKeyBase58 = base58.encode(new Uint8Array(pubPoint.encode()));
```

Comment:
I suspect this odd encoding stuff is what you're talking about. So the previous version of the secp256k1 LD key stuff used `publicKeyWif` as the format. Perhaps we want to look in that...

Here's what the private key wif (wallet import format) looks like, still searching for some text on the public key version:

https://en.bitcoin.it/wiki/Wallet_import_format

Reply:
JFYI, seems like the previous code went here (which suggests it was using a bitcoin address not a public key directly):
https://github.com/bitpay/bitcore-message/blob/master/lib/message.js#L85

Which then goes here to get the public key:
https://github.com/bitpay/bitcore-lib/blob/master/lib/crypto/ecdsa.js#L112-L115